compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -444,9 +444,7 @@ pub fn lower_to_hir(tcx: TyCtxt<'_>, (): ()) -> hir::Crate<'_> {
     sess.time("drop_ast", || drop(krate));
 
     // Discard hygiene data, which isn't required after lowering to HIR.
-    if !sess.opts.unstable_opts.keep_hygiene_data {
-        rustc_span::hygiene::clear_syntax_context_map();
-    }
+    rustc_span::hygiene::clear_syntax_context_map();
```

Comment:
`syntax_context_map` at this point is just a tiny piece of the whole hygiene data, most likely it doesn't make sense to clean it alone.

All hygiene data cannot be cleaned because type-relative name resolution is now hygienic with macros 2.0.
I'm not actually sure if cleaning `syntax_context_map` is correct.

I think it would be better to not clean anything and remove `fn clear_syntax_context_map`.
(And later maybe reconsider which exactly hygiene data can be removed without consequences, and whether it makes sense performance-wise.)

Reply:
Out of curiosity, I did some testing and found that we can clear `expn_hash_to_expn_id`, `syntax_context_map`, and `expn_data_disambiguators` within `clear_syntax_context_map` and all the tests still pass. Clearing the other fields leads to test failures/crashes/etc.

But I'm happy to do your suggestion. Rough profiling on rustc-perf suggests syntax_context_map might be a bit more than 500,000 bytes in a big case (e.g. `serde`).